src/web/mjs/connectors/PeanuToon.mjs

```diff
@@ -0,0 +1,190 @@
+                            async function wait( time ) {
+                                return new Promise( resolve => {
+                                    setTimeout( resolve, time );
+                                } );
+                            };
+                            function getTitleAndUrl(element) {
+                                return {
+                                    url: element.href,
+                                    title: element.text.trim()
+                                };
+                            }; 
+                            function getMangasForMenu(element) {
+                                return ([...document.querySelectorAll("div#로맨스 article div.g-py-10 a")].map(element => getTitleAndUrl(element)));
+                                
+                            };             
+
+
+                            let promise = Promise.resolve();
+                            [...document.querySelectorAll("ul#comic-nav-tabs li.nav-item")].forEach((tab, index) => {
+                                promise = promise.then(async prom => {
+                                    tab.click();
+                                    await wait(1000);
+                                    return getMangasForMenu(tab);
+                                });
+                            });
+
+                            const result = await promise;
+                            resolve(result);
+                                            
+                        }, ${this.config.scrapeDelay.value});`;
+
+        let request = new Request(this.url + '/ko/genre?로맨스', this.requestOptions);
+        return new Promise((resolve, reject) => {
+            return Engine.Request.fetchUI(request, script).then((data) => {
+                const genre_manga = data.map(element => {
+                    return {
+                        id: this.getRootRelativeOrAbsoluteLink(element.url, this.url),
+                        title: element.title
+                    };
+                });
+
+                resolve(genre_manga);
+            }).fail(error => {
+                reject(error);
+            });
+
+        });
+    }
+
+    async _getMangas() {
+
+        let days_manga = await this._getDaysMangas();
+        let genre_manga = await this._getGenreMangas();
+
+        console.log(Array.prototype.concat(days_manga, genre_manga));
+        return Array.prototype.concat(days_manga, genre_manga);
+    }
+
+    async _getChapters(manga) {
+        let request = new Request(this.url + manga.id, this.requestOptions);
+        let data = await this.fetchDOM(request, 'div.detail_area a');
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
+                title:  element.querySelectorAll('div')[7].textContent.trim().split("\n")[0].trim() + " " + element.querySelectorAll('div')[7].textContent.trim().split("\n")[3].trim(),

```

Comment:
Isn't there a more reliable way to get the chapter title? querySelectorAll('div')[7]  and [3] is prone to easy breaking if they change anything in the layout. Better use classes or node attributes in CSS selector if possible.

Reply:
Done!